doc/wg/core/notes/core-notes-2024-10-04.md

```diff
@@ -0,0 +1,123 @@
+ * Lawrence: Yes. I was just worried that the types of enums in userland and kernel could be out-of-sync
+ * Johnathan: There's certainly checking in libtock-rs right now
+ * Leon: Is there any dissenting opinion about these options?
+ * Brad: Not quite a dissent, but I struggle with how we're going to resolve this if we assume we can have pointers anywhere. If a command argument can take a pointer, for instance
+ * Lawrence: I think the plan would be to not allow that. Those must go through allow
+ * Lawrence: Two more places the size pops up is command and upcalls. We were originally focused on return types, but all are worth discussion
+ * Brad: I do think we're all in broad agreement that this should be addressed. But it feels like something more structural here than just choosing types. Maybe I'm not quite right about that or others have an understanding I don't. I think there's a general concept, above and beyond Tock 2.0, and we need to record that wisdom as direction for this change
+ * Lawrence: To clarify, I think the majority thinks that minor changes can't add new type information to the enum. In terms of how this should progress, when should patches land that introduce new types?
+ * Leon: For 2.0, we had a branch. We did one last 1.x release slightly before merging 2.0 into master. I am a little worried about leaving this change in a branch where it bit-rots for a while
+ * Lawrence: Mostly this could be backwards compatible to start, and CHERI could just live in a branch until ready
+ * Leon: I do also think there are a lot of things going on in your PR. Metaptr and raw pointer and things
+ * Lawrence: I think I'll be reducing those
+ * Leon: I don't mean reducing variants, I mean understanding the point of each type for drivers
+ * Lawrence: For CHERI I can be specific. We do want to be more generic though. For CHERI, at any time, userspace might need to be able to access an object at a location, rather than just sending an address. So an authenticated pointer is bigger than u32, but just an address, like a buffer they already own, is usize.
+ * Leon: Do you really ever want addresses without access rights?
+ * Lawrence: Maybe the free interface. Or locations of buffers that applications already own
+ * Johnathan: About metaptr, does it represent capabilities that are pointers only? Aren't there other types of capabilities?
+ * Lawrence: I'm also using it for the register file, so any capabilities.
+ * Johnathan: So it's more than just pointers. Okay. The other thing I'm puzzled on, in non-CHERI you're relying on the cast to copy information.
+ * Lawrence: I wonder if I could rewrite that wrapper layer with strict provenance in mind.
+ * Johnathan: I used `*mut ()` in libtock-rs for similar things. Maybe I'm thinking too strictly about it.
+ * Lawrence: Maybe I should change to strict provenance. Would that cause issues across the kernel?
```

Comment:
@LawrenceEsswood I think you were talking specifically about the strict provenance APIs here, correct? Asking because that slight difference makes my response make sense.

Reply:
I think we did chat about the strict provenance APIs here.

For whatever reason MetaPtr is wrapping a usize even though it could just as easily wrap `*const ()`, and derive from `core::ptr::null()` when it needs to cast from usize to mash a usize into a pointer for argument marshalling reasons.

It would, at the very least, make using the strict provenance API trivial if/when Tock opts in.

Reply:
Okay:

```suggestion
 * Lawrence: Maybe I should change to strict provenance APIs. Would that cause issues across the kernel?
```